src/main/java/org/mule/module/xml/api/SchemaValidationException.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.xml.api;
+
+import static org.mule.module.xml.api.XmlError.SCHEMA_NOT_HONOURED;
+import org.mule.runtime.api.exception.ErrorMessageAwareException;
+import org.mule.runtime.api.message.Message;
+import org.mule.runtime.extension.api.exception.ModuleException;
+
+import java.util.List;
+
+/**
+ * A {@link ModuleException} associated with the {@link XmlError#SCHEMA_NOT_HONOURED} type
+ *
+ * @since 1.0
+ */
+public class SchemaValidationException extends ModuleException implements ErrorMessageAwareException {
+
+  private final Message message;
+
+  /**
+   * Creates a new instance
+   *
+   * @param message  the exception description
+   * @param problems the list of {@link SchemaViolation} found
+   */
+  public SchemaValidationException(String message, List<SchemaViolation> problems) {
+    super(message, SCHEMA_NOT_HONOURED);
+    this.message = Message.builder().collectionValue(problems, SchemaViolation.class).build();
```

Comment:
That would require the ErrorBuilder to be API. It's not API because @afelisatti says that errors should be created in one and only one place inside the runtime. According to her, that's because she has a very particular set of skills. A set of skills that she acquired across time. If I change this, she will use those skills, she will find me, and she will kill me.

Reply:
I'd like to say my words were taken out of context but I won't bother lying. Initially I believed this to be similar to the `validate:all` where errors from other components are aggregated but in this case there are just several internal Java exceptions. So whichever way the data on this is exposed (the exception itself or the error message field) is ok.